### DISCUSSION TITLE
User role check returns false while user has the role

### DISCUSSION THREAD

**Discussion by mohamedds-12** [spatie/laravel-permission#2348] [Q&A] [answered]
If I enabled eager loading of the roles relation in the user model : 
```php
protected $with= ['roles'];
```
When I check if a user has a role or check if he has permission via a role that's assigned to him it returns false.
**Is this acceptable ?**

**Comment by parallels999** [comment, marked as answer]
are you using teams?
let the role relationship be handled by the package

**Reply by mohamedds-12** [reply]
Yes, I'm using teams.
But if I don't do that, I gonna have to call the relation each time I call `User::with('roles')->get()`

**Reply by parallels999** [reply]
if you do `protected $with = ['roles'];`, you are loading roles before team_id is setted, so always gonna be false because it is gonna be empty, 
If you want to do that, you'll have to find another approach, maybe setting team_id on a service provider
also if you want to set team_id manually use `setPermissionsTeamId`
Also maybe this could help https://github.com/spatie/laravel-permission/discussions/2075

**Reply by mohamedds-12** [reply]
Actually, I think it's not worth the trouble, I don't call `User::get()` so often anyway.
Thanks for the help though, Appreciate it.
